tests/tests_pytorch/trainer/connectors/test_accelerator_connector.py

```diff
@@ -429,31 +405,22 @@ def test_strategy_choice_cpu_instance(strategy_class):
         ("ddp", DDPStrategy),
         ("ddp_find_unused_parameters_false", DDPStrategy),
         ("dp", DataParallelStrategy),
-        ("ddp_sharded", DDPShardedStrategy),
-        ("ddp_sharded_spawn", DDPSpawnShardedStrategy),
         pytest.param("deepspeed", DeepSpeedStrategy, marks=RunIf(deepspeed=True)),
     ],
 )
-def test_strategy_choice_gpu_str(strategy, strategy_class):
-    if "sharded" in strategy:
-        with pytest.deprecated_call(match="FairScale has been deprecated in v1.9.0"):
-            trainer = Trainer(strategy=strategy, accelerator="gpu", devices=2)
-    else:
-        trainer = Trainer(strategy=strategy, accelerator="gpu", devices=2)
+def test_strategy_choice_gpu_str(strategy, strategy_class, cuda_count_2):
+    trainer = Trainer(strategy=strategy, accelerator="gpu", devices=2)
     assert isinstance(trainer.strategy, strategy_class)
 
 
-@RunIf(min_cuda_gpus=2)
 @pytest.mark.parametrize("strategy_class", [DDPSpawnStrategy, DDPStrategy])
-def test_strategy_choice_gpu_instance(strategy_class):
+def test_strategy_choice_gpu_instance(strategy_class, cuda_count_2):
     trainer = Trainer(strategy=strategy_class(), accelerator="gpu", devices=2)
     assert isinstance(trainer.strategy, strategy_class)
```

Comment:
When you do these changes, make sure not to forget the MPS case. Even if cuda_count_2 is set, MPS has precedence in the connector. 

In such cases, you could consider adding mps_count_0 or switch accelerator="cuda" when appropriate.